src/jit/rationalize.cpp

```diff
@@ -1903,31 +1667,6 @@ Compiler::fgWalkResult Rationalizer::SimpleTransformHelper(GenTree **ppTree, Com
         JITDUMP("\n");
         return SimpleTransformHelper(ppTree, data);
     }
-    else if (tree->gtOper == GT_QMARK)
-    {
-        // only certain forms of qmarks are allowed
-        // qmark(conditionExpr, 1, 0) is equivalent to conditionExpr
```

Comment:
Compiler::fgCheckQmarkAllowedForm() - is asserting for RyuJIT back-end no qmarks beyond morph in debug builds.  But in release, 0/1/ qmark expansion would still happen in rationalizer, if there are any. Post this change, any QMarks should hit a noway assert in lower/codegen?